tests/console/java.pm

```diff
@@ -21,56 +21,41 @@ use base "consoletest";
     pkcon_quit;
-
-    if (check_var("DISTRI", "sle")) {
-        if (zypper_call('in --auto-agree-with-licenses java-*', timeout => 500, exitcode => [0, 4])) {
-            # install only java-11-openjdk* & java-*-ibm*
-            zypper_call('in --auto-agree-with-licenses java-11-openjdk* java-*-ibm*');
-        }
-    }
-
-    if (check_var("DISTRI", "opensuse")) {
-        # Capture the return code value of the following scenarios
-        my $bootstrap_pkg_rt       = zypper_call("se java-*bootstrap",                             exitcode => [0, 104]);
-        my $bootstrap_conflicts_rt = zypper_call("in --auto-agree-with-licenses --dry-run java-*", exitcode => [0, 4]);
-
-        # logs / debugging purposes
-        diag "checking variable: bootstrap_pkg_rt = $bootstrap_pkg_rt";
-        diag "checking variable: bootstrap_conflicts_rt = $bootstrap_conflicts_rt";
-
-        if ($bootstrap_pkg_rt == 0) {
-            diag "There are java bootstrap packages available to be installed";
-            print "There are java bootstrap packages available to be installed\n";
-            if ($bootstrap_conflicts_rt == 0) {
-                diag "There is no conflict installing the java bootstrap packages";
-                print "There is no conflict installing the java bootstrap packages\n";
-                zypper_call("in java-*");
-            }
-            else {
-                diag "There are conflicts with the installation of java bootstrap packages";
-                print "There are conflicts with the installation of java bootstrap packages\n";
-                record_soft_failure 'boo#1019090';
-                # Workaround: install java-* except from the problematic bootstrap packages
-                zypper_call "in `(zypper se java-* | grep -v bootstrap | grep -v 'i ' | awk '{print \$2}' | sed -n -E -e '/java/,\$ p')`";
-            }
-        }
-        else {
-            diag "There are no java bootstrap packages";
-            print "There are no java bootstrap packages\n";
-            zypper_call("in java-*");
-        }
+    # if !QAM test suite then register Legacy module
+    (!!is_updates_tests || is_opensuse) || add_suseconnect_product('sle-module-legacy');
+    # Supported Java versions for sle15sp2
+    # https://www.suse.com/releasenotes/x86_64/SUSE-SLES/15-SP2/#development-java-versions
+    # java-11-openjdk                   -> Basesystem
+    # java-10-openjdk & java-1_8_0-ibm  -> Legacy
+    my $cmd = 'install --auto-agree-with-licenses ';
+    $cmd .= (is_sle('15+') || is_leap) ? 'java-11-openjdk* java-1_*' : 'java-*';
+    zypper_call($cmd);
+
+    if (script_run 'rpm -q wget') {
+        zypper_call 'in wget';
     }
-
-    zypper_call 'in wget';
     assert_script_run 'wget --quiet ' . data_url('console/test_java.sh');
     assert_script_run 'chmod +x test_java.sh';
-    assert_script_run './test_java.sh';
+    my $pipe_fail = testapi::is_serial_terminal() ? 'bash -oe pipefail;' : 'bash -eox pipefail;';
```

Comment:
With the ';' it'll just start a new shell and wait for input definitely.

Reply:
In consoletest_setup, it already does `echo "set -o pipefail" >> /etc/bash.bashrc.local`, that might be enough?

Reply:
ah, I thought it just separates commands. thanks for hint

Reply:
Yes, exactly. First it starts `bash -oe pipefail`, which waits for input and after that exits, the parent shell runs the script.

Reply:
It should be enough as long as the test suite does not reboot VM.

```
    script_run 'echo "set -o pipefail" >> /etc/bash.bashrc.local';
    script_run '. /etc/bash.bashrc.local';
```

I haven't found ` /etc/bash.bashrc.local` in bash man though. Only `/etc/bash.bashrc ` ...

Reply:
> I haven't found /etc/bash.bashrc.local in bash man though. Only /etc/bash.bashrc

That's sourced by `/etc/bash.bashrc`. So as long as the serial terminal is opened after `consoletest_setup` applied the change, pipefail should be active.